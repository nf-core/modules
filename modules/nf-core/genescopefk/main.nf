process GENESCOPEFK {
    tag "$meta.id"
    label 'process_low'

    if (params.enable_conda) {
        error "Conda environments cannot be used when using the GeneScope tool. Please use docker or singularity containers."
    }
    // WARN: Version information not provided by tool on CLI. Please update version string below when bumping container versions.
    container 'ghcr.io/nbisweden/fastk_genescopefk_merquryfk:1.2'

    input:
    tuple val(meta), path(fastk_histex_histogram)

    output:
    tuple val(meta), path("*_linear_plot.png")            , emit: linear_plot
    tuple val(meta), path("*_log_plot.png")               , emit: log_plot
    tuple val(meta), path("*_model.txt")                  , emit: model
    tuple val(meta), path("*_summary.txt")                , emit: summary
    tuple val(meta), path("*_transformed_linear_plot.png"), emit: transformed_linear_plot
    tuple val(meta), path("*_transformed_log_plot.png")   , emit: transformed_log_plot
    path "versions.yml"                                   , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def GENESCOPE_VERSION = '380815c420f50171f9234a0fd1ff426b39829b91' // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
    """
    GeneScopeFK.R \\
        $args \\
        --input $fastk_histex_histogram \\
        --output . \\
        --name_prefix ${prefix}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        genescope: $GENESCOPE_VERSION
        r: \$( R --version | sed '1!d; s/.*version //; s/ .*//' )
    END_VERSIONS
    """
}
