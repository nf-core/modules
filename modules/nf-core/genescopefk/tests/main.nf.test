nextflow_process {

    name "Test Process GENESCOPEFK"
    config "./nextflow.config"
    script "../main.nf"
    process "GENESCOPEFK"

    tag "modules"
    tag "modules_nfcore"
    tag "fastk"
    tag "fastk/fastk"
    tag "fastk/histex"
    tag "genescopefk"

    setup {
        run("FASTK_FASTK") {
            script "../../fastk/fastk/main.nf"

            process {
                """
                input[0] = [
                    [id:'test', single_end:true],// meta map
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/fastq/test_1.fastq.gz',checkIfExists:true)
                ]
                """
            }
        }

        run("FASTK_HISTEX") {
            script "../../fastk/histex/main.nf"

            process {
                """
                input[0] = FASTK_FASTK.out.hist
                """
            }
        }
    }

    test("homo_sapiens - hist") {
        when {
            params {
                fastk_histex_arg = "-G"
                genescopefk_arg  = "--kmer_length 40 --ploidy 1"
            }
            process {
                """
                input[0] = FASTK_HISTEX.out.hist
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("homo_sapiens - hist -- stub") {
        options '-stub'
        when {
            params {
                fastk_histex_arg = "-G"
                genescopefk_arg  = "--kmer_length 40 --ploidy 1"
            }
            process {
                """
                input[0] = FASTK_HISTEX.out.hist
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out,
                    path(process.out.versions[0]).yaml
                ).match() }
            )
        }
    }
}