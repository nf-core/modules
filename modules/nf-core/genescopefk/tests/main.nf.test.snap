{
    "homo_sapiens - hist": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        [
                            "test_linear_plot.png:md5,bb221e2e21a6c0d04dcb6ec8f1bf1cd7",
                            "test_transformed_linear_plot.png:md5,3c8e3ad315e6c15cdf51c87de277ec1f"
                        ]
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        [
                            "test_log_plot.png:md5,d71ad0c963775992043c8f3c55d27fc4",
                            "test_transformed_log_plot.png:md5,90d983c6faa9b8a02a1d26b777feefc6"
                        ]
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test_model.txt:md5,49e5a66cf02b53a3b20880bfabef2ee8"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test_summary.txt:md5,34540dcc2ba4ab9530f8878bbdc0921c"
                    ]
                ],
                "4": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test_transformed_linear_plot.png:md5,3c8e3ad315e6c15cdf51c87de277ec1f"
                    ]
                ],
                "5": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test_transformed_log_plot.png:md5,90d983c6faa9b8a02a1d26b777feefc6"
                    ]
                ],
                "6": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "12.25"
                    ]
                ],
                "7": [
                    "versions.yml:md5,de2e6adb831619dff8e2e7262736ba5a"
                ],
                "kmer_cov": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "12.25"
                    ]
                ],
                "linear_plot": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        [
                            "test_linear_plot.png:md5,bb221e2e21a6c0d04dcb6ec8f1bf1cd7",
                            "test_transformed_linear_plot.png:md5,3c8e3ad315e6c15cdf51c87de277ec1f"
                        ]
                    ]
                ],
                "log_plot": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        [
                            "test_log_plot.png:md5,d71ad0c963775992043c8f3c55d27fc4",
                            "test_transformed_log_plot.png:md5,90d983c6faa9b8a02a1d26b777feefc6"
                        ]
                    ]
                ],
                "model": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test_model.txt:md5,49e5a66cf02b53a3b20880bfabef2ee8"
                    ]
                ],
                "summary": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test_summary.txt:md5,34540dcc2ba4ab9530f8878bbdc0921c"
                    ]
                ],
                "transformed_linear_plot": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test_transformed_linear_plot.png:md5,3c8e3ad315e6c15cdf51c87de277ec1f"
                    ]
                ],
                "transformed_log_plot": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test_transformed_log_plot.png:md5,90d983c6faa9b8a02a1d26b777feefc6"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,de2e6adb831619dff8e2e7262736ba5a"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-04-11T15:34:21.631389751"
    },
    "homo_sapiens - hist -- stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        [
                            "test_linear_plot.png:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test_transformed_linear_plot.png:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        [
                            "test_log_plot.png:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test_transformed_log_plot.png:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test_model.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test_summary.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "4": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test_transformed_linear_plot.png:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "5": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test_transformed_log_plot.png:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "6": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "0.00"
                    ]
                ],
                "7": [
                    "versions.yml:md5,de2e6adb831619dff8e2e7262736ba5a"
                ],
                "kmer_cov": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "0.00"
                    ]
                ],
                "linear_plot": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        [
                            "test_linear_plot.png:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test_transformed_linear_plot.png:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "log_plot": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        [
                            "test_log_plot.png:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test_transformed_log_plot.png:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "model": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test_model.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "summary": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test_summary.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "transformed_linear_plot": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test_transformed_linear_plot.png:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "transformed_log_plot": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test_transformed_log_plot.png:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,de2e6adb831619dff8e2e7262736ba5a"
                ]
            },
            {
                "GENESCOPEFK": {
                    "genescopefk": "380815c420f50171f9234a0fd1ff426b39829b91",
                    "r": "4.2.0"
                }
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-04-11T15:05:55.106012575"
    }
}